pkg/plugins/gateway/util.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2024 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gateway
+
+import (
+	"encoding/json"
+	"slices"
+	"strings"
+
+	configPb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	extProcPb "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	envoyTypePb "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/vllm-project/aibrix/pkg/utils"
+	"k8s.io/klog/v2"
+)
+
+// validateStreamOptions validates whether stream options to include usage is set for user request
+func validateStreamOptions(requestID string, user utils.User, jsonMap map[string]interface{}) *extProcPb.ProcessingResponse {
```

Comment:
How can we know user are using heterogenous features?  If we have the feature flag, we can do some validation here as well. not something we need to fix in this PR, just surface this requirement

Reply:
I am thinking we should have a feature flag for heterogenous feature. Right now it is enabled by default and incurs some performance penalty.

Reply:
yeah, please help document this issue.